src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpointManagerBase.cs

```diff
@@ -214,6 +214,39 @@ private Task RenameHubServiceEndpoint(HubServiceEndpoint endpoint)
             return Task.CompletedTask;
         }
 
+        private void UpdateNegotiationEndpointsStore(IReadOnlyList<HubServiceEndpoint> endpoints, ScaleOperation scaleOperation)
+        {
+            foreach (var hubEndpoint in _endpointsPerHub)
+            {
+                var updatedEndpoints = endpoints.Where(e => e.Hub == hubEndpoint.Key).ToList();
+                var oldEndpoints = hubEndpoint.Value;
+                var newEndpoints = new List<HubServiceEndpoint>();
+                switch (scaleOperation)
+                {
+                    case ScaleOperation.Rename:
+                        newEndpoints = oldEndpoints.Except(updatedEndpoints, new HubServiceEndpointWeakComparer()).ToList();
+                        newEndpoints.AddRange(updatedEndpoints);
```

Comment:
>newEndpoints [](start = 24, length = 12)

there is no need to propagate the change to MultiEndpointContainer, if the container holds the same `HubServiceEndpoint` object

Reply:
Good catch. Need to update the logic a little to avoid HubServiceEndpoint total replacement in ServiceEndpointManager side, so that the logic in container could be removed entirely for rename part.